packages/core/test/id.test.ts

```diff
@@ -108,4 +109,37 @@ describe('State node IDs', () => {
       }
     });
   });
+
+  it('should work with IDs that have escaped periods', () => {
```

Comment:
I don't think this test title is accurate - this test should be renamed (and maybe even moved?). This focuses on a state key that contains such escaped dots. 

A test case of this exact test title is needed though but it's a different test case:
```ts
    const machine = createMachine({
      initial: 'start',
      states: {
        start: {
          on: {
            escaped: '#xyz\\.bar',
            unescaped: '#xyz.bar2'
          }
        },
        foo: {
          id: 'xyz',
          initial: 'bar1',
          states: {
            bar1: {},
            bar2: {}
          }
        },
        baz: {
          id: 'xyz\\.bar'
        },
      }
    });
```